### PR TITLE
Add the ability for site files to be served as custom content-types

### PR DESCRIPTION
**Pull request by trevorb**
Templates that are being served as text/html are being processed by varnish and rewritten which is causing issues.  We should really be serving these templates as text/plain anyways but this should fix the issue.

### CODE REVIEW

File: pegasus/router.rb [hunk 1]
```diff
@@ -288,6 +288,9 @@ def self.load_configs_in(dir)
       response.headers['X-Frame-Options'] = 'ALLOWALL'
     end
 
+    if @locals[:header]['content-type']
```
Comment (davidsbailey) [inline pegasus/router.rb]: Apologies for my Pegasus ignorance, but does @locals[:header] refer only to the block at the beginning of a haml file such as the one you've modified below in records.haml?

Comment (trevorb) [inline pegasus/router.rb]: Yep, only the header section of the haml file

**Comment by davidsbailey** [comment]
Great. Looks good to me!

**Comment by wjordan** [comment]
Can you provide more specifics on what exactly the issues we're experiencing are and how to reproduce them, and how this change fixes it? To my knowledge Varnish does not do any processing or rewriting of html files, so I suspect that something else is going on here. I'd like to figure out exactly what so that this doesn't continue to bite us in the future.

Also, I don't quite follow why should we be serving this page as `text/plain` anyways? Unless I'm missing something it looks like this `haml` page is generating HTML content, which should be served as `text/html`.

**Comment by trevorb** [comment]
Sure.  The issue was originally exposed by the data import feature of applab breaking.  (https://code.org/edit-csp-table/zej8xh1FMbgNxILX5rjvcg/test).  After some research I tracked it down to the angular html template being escaped in a way that it shouldn't.  A variable substitution line of the template was being url escaped.  The issue only happens in production / staging and not in development.

I logged into a production frontend machine and did a curl directly to the rails server and the html template was being returned correctly.  However, if you request the same template from code.org instead of directly from the rails server the variable substitution line in the html template is escaped and incorrect.

You can verify this by following these steps:
1. SSH into a production frontend machine
2. Make a request to the rails server.  Notice that {{'/v3/import-shared-tables/' + channelId + '/' + tableName}} is not escaped.
$ curl -v localhost:8081/edit-csp-table/records

This returns:
  # A BUNCH OF HTML HERE
      <form accept='.csv' action="{{'/v3/import-shared-tables/' + channelId + '/' + tableName}}" enctype='multipart/form-data' method='POST' ng-show='importing'>
      # MORE HTML HERE

3. Make a request to code.org.  Notice that the line is url escaped "%20+%20channelId%20+%20'/'%20+%20tableName%7D%7D"

$ curl -v https://code.org/edit-csp-table/records

This returns:
  # A BUNCH OF HTML HERE
      <form accept=".csv" action="%7B%7B'/v3/import-shared-tables/'%20+%20channelId%20+%20'/'%20+%20tableName%7D%7D" enctype="multipart/form-data" method="POST" ng-show="importing">
      # MORE HTML HERE



To directly answer your question about the content type.  The haml file is generating an html template that angular uses to render html and shouldn't really be rendered as html.

**Comment by davidsbailey** [comment]
Excellent point Will. https://github.com/angular/angular.js/issues/6860 suggests that text/ng-template might be the right content-type to set.

**Comment by trevorb** [comment]
Cool, I just updated to use text/ng-template as the content-type

**Comment by wjordan** [comment]
ah ok, thanks for the clarification that this is rendering an Angular template and not a straight html page. Yeah, I think setting the content-type to `text/ng-template` seems like the right fix here.

Based on the description of the issue, it sounds like [UpgradeInsecureRequests](https://github.com/code-dot-org/code-dot-org/blob/staging/lib/cdo/rack/upgrade_insecure_requests.rb) (which only runs on HTTPS requests, which is why it only affected requests to `code.org` and not from the server directly) was parsing/filtering the returned content as though it were a valid HTML document, and since the content was not a valid HTML document but rather an Angular template, it caused the unexpected escaping issue. So the fix looks good to me!

**Comment by davidsbailey** [comment]
FYI @CaptainStack and @laurelfan in case you run into a similar problem in ops-dashboard or teacher-dashboard.

**Pull request merged**